docs/source/isolation.rst

```diff
@@ -0,0 +1,200 @@
+Master mode, it is responsible for communicating between agent processes and
+frameworks. A Framework is a program which wants to run tasks on the Mesos
+cluster.
+
+A master is responsible for presenting frameworks with resource offers.
+Resource offers are compute resource free for a framework to run a task. The
+details of that compute resource comes from the agent nodes, which regularly
+tell the Master agent the resources it has available for running tasks. Using
+the correct parlance, Mesos agents make 'offers' to the master.
+
+At Yelp, the resource that a given agent has available is determined by a
+couple of things: if the node also runs 'classic' services, then the resources
+given to the agent are decided by a script which takes a rough guess at how
+much resource is used *without* Mesos running, and subtracts it from the total
+available resource. That gives a *very* approximate guess at the resources
+available for the Mesos Agent to run tasks.
+
+If, however, the agent node is a dedicated PaaSTA box (defined by having the
+puppet role ``paasta``), then all the resources available on the box are given
+to the Agent to run tasks. This means that the cpus offered is set to the
+total number of processors on the host, and the memory offered is equal to the
+total memory available on the host.
+
+Once a master node receives offers from an agent, it forwards it to
+a framework. Resource offers are split between frameworks according to
+the master's configuration - there may be particular priority given
+to some frameworks.
+
+At Yelp, we treat the frameworks we run (at the time of writing, Marathon and
+Chronos) equally. That means that frameworks *should* have offers distributed
+between them evenly, and all tasks are considered equal.
+
+It is then up to the framework to decide what it wants to do with an offer.
+The framework may decide to:
+
+  * Reject the offer, if the framework has no tasks to run.
+  * Reject the offer, if the resources included in the offer are not enough to
+    match those required by the application.
+  * Reject the offer, if attributes on the slave conflict with any constraints
+    set by the task.
+  * Accept the offer, if there is a task that requires resources less than or
+    equal to the resources offered by the Agent.
+
+When rejecting an offer, the framework may apply a 'filter' to the offer. This
+filter is then used by the Mesos master to ensure that it does *not* resend
+offers that are 'filtered' by a framework. The default filter applied includes
+a timeout - a Master will not resend an offer to a framework for a period of 5
+seconds.
+
+If a framework decides it wants to accept a resource offer, it then tells the
+master to run a task on the agent. The details of the 'acceptance' include a
+detail of the task to be run, and the 'executor' used to run the task.
+
+By default, Yelp uses the 'Docker' executor everywhere. This means that *all*
```

Comment:
This you can say "paasta", docker is hard-coded into how paasta creates these things